enhancements/authentication/external-oidc-additional-identity-information-sources.md

```diff
@@ -0,0 +1,804 @@
+correct components based on the user-provided configuration in the `authentications.config.openshift.io/cluster`
+resource.
+
+If the configuration does not configure any providers with the new external claim sourcing configuration options,
+it will ensure that the webhook authenticator is not deployed and will directly configure the Kubernetes API server
+using the Structured Authentication Configuration file.
+
+If the configuration only contains providers that use the new external claim sourcing configuration options,
+it will ensure that the webhook authenticator is deployed and configured, the Kubernetes API server is configured
+to use the webhook authenticator, and the Structured Authentication Configuration is not configured on the Kubernetes API server
+
+If there is a mixed configuration, it will ensure that the webook authenticator is deployed and configured for the providers leveraging
+the new configuration options, the Kubernetes API server is configured to use the webhook authenticator, and the Structured Authentication
+Configuration is configured on the Kubernetes API server for the providers that are not configured to leverage the new functionality.
+
+Diagram of the new behavior:
+```mermaid
+flowchart LR
+    A("Cluster Authentication Operator") -- Reads --> B
+    B("authentications.config.openshift.io/cluster")
+    C{"External Claim Sources Configured?"}
+    A --> C
+    C -- Yes --> D
+    C -- No --> F
+    D{"Only External Claim Sources Configured?"}
+    D -- Yes --> E
+    D -- No --> E
+    D -- No --> F
+    E("Deploy + Configure Webhook Authenticator")
+    F("Configure kube-apiserver Structured Authn.")
+```
+
+##### Full Architecture Diagram
+
+```mermaid
+flowchart TD
+    subgraph Cluster
+        subgraph kube-apiserver
+            subgraph Authentication
+                A("...") -- Chains --> B
+                B("JWTAuthenticator [Structured Authn]") -- Chains --> C
+                C("WebhookTokenAuthenticator")
+            end
+
+            Authentication -- Sends UserInfo --> D
+            D("Authorization")
+        end
+
+        E("authentications.config.openshift.io/cluster")
+        F("Cluster Authentication Operator") -- Reads --> E
+        F -- Configures --> B
+        F -- Configures --> C
+        F -- Deploys --> G
+        G("New Webhook Authenticator")
+        C -- TokenRequest --> G
+        G -- UserInfo --> C
+    end
+
+    H("External Claim Source")
+    G -- Fetches Claims --> H
+
+    I("Request") --> A
+```
+
+### Risks and Mitigations
+
+1. Misconfiguration of external claim sources
+    * **Mitigation**: Perform robust admission time validation as we already do on the `Authentication` API.
+    CEL expressions will have to successfully compile and will be limited in their length to prevent excessive
+    compilation and run times.
+
+2. Introduction of network latency to authentication
```

Comment:
IIUC the webhook authn plugin maintains a cache, too, which should help. Does there need to be any kind of size limitation on TokenReview responses?